### ISSUE TITLE
Can not get duration of video .mkv

### ISSUE THREAD

**Issue by thieuyd** [pascalbaljetmedia/laravel-ffmpeg#17]
Dear all.

I can not get duration of video file type = .mkv('mkv' => 'video/x-matroska',)

File Media.php, function getDurationInSeconds() Line 34. 

We don't have key 'duration' so i can not get duration time of video. How can i get duration time of video file type = .mkv.
I don't have proplem with other file type (avi, mp4, mpeg, mov...)

Function:
public function getDurationInSeconds(): int
{
    return $this->media->getStreams()->first()->get('duration');
}

Result of  dd($this->media->getStreams()->first())
Stream {#455
  -properties: array:34 [
    "index" => 0
    "codec_name" => "h264"
    "codec_long_name" => "H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10"
    "profile" => "High"
    "codec_type" => "video"
    "codec_time_base" => "317/19001"
    "codec_tag_string" => "[0][0][0][0]"
    "codec_tag" => "0x0000"
    "width" => 1920
    "height" => 1080
    "coded_width" => 1920
    "coded_height" => 1080
    "has_b_frames" => 1
    "sample_aspect_ratio" => "1:1"
    "display_aspect_ratio" => "16:9"
    "pix_fmt" => "yuv420p"
    "level" => 40
    "color_range" => "tv"
    "color_space" => "bt709"
    "color_transfer" => "bt709"
    "color_primaries" => "bt709"
    "chroma_location" => "left"
    "field_order" => "progressive"
    "refs" => 1
    "is_avc" => "true"
    "nal_length_size" => "4"
    "r_frame_rate" => "19001/634"
    "avg_frame_rate" => "19001/634"
    "time_base" => "1/1000"
    "start_pts" => 0
    "start_time" => "0.000000"
    "bits_per_raw_sample" => "8"
    "disposition" => array:12 [
      "default" => 1
      "dub" => 0
      "original" => 0
      "comment" => 0
      "lyrics" => 0
      "karaoke" => 0
      "forced" => 0
      "hearing_impaired" => 0
      "visual_impaired" => 0
      "clean_effects" => 0
      "attached_pic" => 0
      "timed_thumbnails" => 0
    ]
    "tags" => array:1 [
      "language" => "eng"
    ]
  ]
}

Thanks & best regards.

**Comment by pascalbaljet** [comment]
What is the result of `dd($this->media->getStreams())`?

**Comment by thieuyd** [comment]
We don't have property duration in the result of getStreams()
Result of dd($this->media->getStreams())
StreamCollection {#458
  -streams: array:1 [
    0 => Stream {#455
      -properties: array:30 [
        "index" => 0
        "codec_name" => "h264"
        "codec_long_name" => "H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10"
        "profile" => "High"
        "codec_type" => "video"
        "codec_time_base" => "1/60"
        "codec_tag_string" => "[0][0][0][0]"
        "codec_tag" => "0x0000"
        "width" => 1280
        "height" => 768
        "coded_width" => 1280
        "coded_height" => 768
        "has_b_frames" => 2
        "sample_aspect_ratio" => "1:1"
        "display_aspect_ratio" => "5:3"
        "pix_fmt" => "yuv420p"
        "level" => 51
        "chroma_location" => "left"
        "field_order" => "progressive"
        "refs" => 1
        "is_avc" => "true"
        "nal_length_size" => "4"
        "r_frame_rate" => "30/1"
        "avg_frame_rate" => "30/1"
        "time_base" => "1/20000"
        "start_pts" => 0
        "start_time" => "0.000000"
        "bits_per_raw_sample" => "8"
        "disposition" => array:12 [
          "default" => 1
          "dub" => 0
          "original" => 0
          "comment" => 0
          "lyrics" => 0
          "karaoke" => 0
          "forced" => 0
          "hearing_impaired" => 0
          "visual_impaired" => 0
          "clean_effects" => 0
          "attached_pic" => 0
          "timed_thumbnails" => 0
        ]
        "tags" => array:1 [
          "language" => "eng"
        ]
      ]
    }
  ]
}

**Comment by thieuyd** [comment]
Dear all.

After I check code. I want to edit function getDurationInSeconds() line 34 File Media.php.

My function :
```
public function getDurationInSeconds(): int
    {
        $duration = $this->media->getStreams()->first()->get('duration');

        if (empty($duration)) {
            $duration = $this->media->getFormat()->get('duration');
        }

        if (empty($duration)) {
            $duration = 0;
        }

        return $duration;
    }
```

**Comment by pascalbaljet** [comment]
So `$this->media->getFormat()->get('duration')` returns the right duration of the video?

**Comment by thieuyd** [comment]
1. Yes.
2. Other one. The function getDurationInSeconds return type of value is int. But result some time don't return int. My problem is a example. It return null.
You can test with command "ffprobe -v error -show_format -show_streams mkv.mkv"

**Comment by pascalbaljet** [comment]
Thanks for the investigation! I'll leave this issue open until I've written a fix for it.

**Comment by pascalbaljet** [comment]
Fixed in [1.1.7](https://github.com/pascalbaljetmedia/laravel-ffmpeg/releases/tag/1.1.7)
